nibabel/cifti2/cifti2_axes.py

```diff
@@ -0,0 +1,1330 @@
+
+        for check_name in ('name', 'voxel', 'vertex'):
+            shape = (self.size, 3) if check_name == 'voxel' else (self.size, )
+            if getattr(self, check_name).shape != shape:
+                raise ValueError("Input {} has incorrect shape ({}) for Label axis".format(
+                        check_name, getattr(self, check_name).shape))
+
+    @classmethod
+    def from_mask(cls, mask, name='other', affine=None):
+        """
+        Creates a new BrainModel axis describing the provided mask
+
+        Parameters
+        ----------
+        mask : np.ndarray
+            all non-zero voxels will be included in the BrainModel axis
+            should be (Nx, Ny, Nz) array for volume mask or (Nvertex, ) array for surface mask
+        name : str
+            Name of the brain structure (e.g. 'CortexRight', 'thalamus_left' or 'brain_stem')
+        affine : np.ndarray
+            (4, 4) array with the voxel to mm transformation (defaults to identity matrix)
+            Argument will be ignored for surface masks
+
+        Returns
+        -------
+        BrainModel which covers the provided mask
+        """
+        if affine is None:
+            affine = np.eye(4)
+        if np.asarray(affine).shape != (4, 4):
+            raise ValueError("Affine transformation should be a 4x4 array or None, not %r" % affine)
+        if mask.ndim == 1:
+            return cls.from_surface(np.where(mask != 0)[0], mask.size, name=name)
+        elif mask.ndim == 3:
+            voxels = np.array(np.where(mask != 0)).T
+            return cls(name, voxel=voxels, affine=affine, volume_shape=mask.shape)
+        else:
+            raise ValueError("Mask should be either 1-dimensional (for surfaces) or "
+                             "3-dimensional (for volumes), not %i-dimensional" % mask.ndim)
+
+    @classmethod
+    def from_surface(cls, vertices, nvertex, name='Cortex'):
+        """
+        Creates a new BrainModel axis describing the vertices on a surface
+
+        Parameters
+        ----------
+        vertices : np.ndarray
+            indices of the vertices on the surface
+        nvertex : int
+            total number of vertices on the surface
+        name : str
+            Name of the brain structure (e.g. 'CortexLeft' or 'CortexRight')
+
+        Returns
+        -------
+        BrainModel which covers (part of) the surface
+        """
+        cifti_name = cls.to_cifti_brain_structure_name(name)
+        return cls(cifti_name, vertex=vertices,
+                   nvertices={cifti_name: nvertex})
+
+    @classmethod
+    def from_mapping(cls, mim):
+        """
+        Creates a new BrainModel axis based on a CIFTI dataset
+
+        Parameters
+        ----------
+        mim : cifti2.Cifti2MatrixIndicesMap
+
+        Returns
+        -------
+        BrainModel
+        """
+        nbm = np.sum([bm.index_count for bm in mim.brain_models])
+        voxel = -np.ones((nbm, 3))
```

Comment:
Given that these are going to be cast to `int`, later:

```suggestion
        voxel = np.full((nbm, 3), fill_value=-1, dtype=int)
```